### PR TITLE
Remove runtime unit test dependency on aoc

### PR DESCRIPTION
**Pull request by sophimao**
Runtime unit tests testing `clBuildProgram` on programs created from source would invoke the aoc compiler when the context compiler mode is not `CL_CONTEXT_COMPILER_MODE_OFFLINE_INTELFPGA`. This is not ideal as we would want the unit tests to be standalone from the compiler as much as possible.

This change removes this dependency on the aoc compiler by adding precompiled binaries to the repository to be loaded directly by the affected unit tests. The scope of the change is restricted to the unit tests, the runtime functionality related to building from source remains the same as before.

With that being said, based on the [OpenCL spec](https://registry.khronos.org/OpenCL/sdk/3.0/docs/man/html/clBuildProgram.html), if the program is created with `clCreateProgramWithSource` and `CL_DEVICE_COMPILER_AVAILABLE` is set to false, then `clBuildProgram` should return `CL_COMPILER_NOT_AVAILABLE`. This justifies us to remove the support for compiling source kernel code just-in-time, so we will be able to clean-up the code that involves the aoc compiler even more, and at the time we completely remove the support we can just remove the `from_source` tests, which should allow us to clean up some of the added precompiled binaries in this change as well.

This change also updates the runtime container images to no longer install aoc. Newly built container images, on which aoc is not installed, is verified to be working with the new runtime unit tests in my forked repository. [Official containers](https://github.com/orgs/intel/packages?q=fpga_runtime&tab=packages&q=*-dev) built starting from this change will no longer contain aoc.

### CODE REVIEW

File: container/ubuntu-22.04-dev/Dockerfile [hunk 1]
```diff
@@ -40,14 +40,12 @@ RUN apt-get -y update \
...
   && useradd --system ${uid:+--uid "$uid"} --user-group --shell /sbin/nologin --create-home --home-dir /home/build build \
   && echo 'build ALL=(ALL) NOPASSWD:ALL' > /etc/sudoers.d/build \
   && rm -rf "$PWD"
 
 USER build
 WORKDIR /home/build
 
-ENV PATH="/opt/aocl/hld/bin:/usr/lib/llvm-13/bin:$PATH"
-RUN aoc -version
+ENV PATH="/usr/lib/llvm-13/bin:$PATH"
```
[the remark this answers is not in the document]
Comment (sophimao) [reply container/ubuntu-22.04-dev/Dockerfile:50]: I believe it is for clang-format, we run that in CI

File: test/acl_program_test.cpp [hunk 2]
```diff
@@ -1049,8 +1068,27 @@ MT_TEST(from_source, online_mode) {
   // Actually run the kernel compiler command.
   // Pretend we have an online compiler
   unload();
-  load(CL_CONTEXT_COMPILER_MODE_ONLINE_INTELFPGA, "aoc -rtl -v --no-env-check",
-       false);
+
```
[the remark this answers is not in the document]
Comment (mendell27) [reply test/acl_program_test.cpp:1071]: yes

File: test/acl_program_test.cpp [hunk 3]
```diff
@@ -1082,11 +1099,9 @@ MT_TEST(from_source, online_mode) {
   ACL_LOCKED(CHECK(acl_realpath_existing(check_str) != ""));
   check_str = m_hashdir + std::string("/build.cmd");
   ACL_LOCKED(CHECK(acl_realpath_existing(check_str) != ""));
-  // Since we invoked compiler with -rtl, we only get an .aoco and .aocr
+  // If we have invoked compiler with -rtl, we should get an .aocr
   check_str = m_hashdir + std::string("/kernels.aocr");
   ACL_LOCKED(CHECK(acl_realpath_existing(check_str) != ""));
-  check_str = m_hashdir + std::string("/kernels/kernels.v");
```
Comment (sophimao) [inline test/acl_program_test.cpp:1088]: Removed this check as we are now just emulating the compile by copying over the precompiled aocr file, and not producing the hardware IP.

**Review by mendell27 — DISMISSED** [review]

**Comment by sophimao** [comment]
Did a rebase to clean up commits

**Review by mendell27 — APPROVED** [review]

**Pull request merged**